src/ngx_http_vhost_traffic_status_shm.c

```diff
@@ -520,6 +558,27 @@ ngx_http_vhost_traffic_status_shm_add_upstream(ngx_http_request_t *r)
     return NGX_OK;
 }
 
+static int ngx_http_vhost_traffic_status_find_status_code_slot_cmp(const void *one, const void *two)
+{
+    return (*(ngx_uint_t *) one - *(ngx_uint_t *) two);
+}
+
+ngx_int_t
+ngx_http_vhost_traffic_status_find_status_code_slot(ngx_uint_t status, ngx_array_t *status_codes)
+{
+    ngx_uint_t *found = (ngx_uint_t *) bsearch(&status, status_codes->elts, status_codes->nelts,
+        sizeof(ngx_uint_t),
+        ngx_http_vhost_traffic_status_find_status_code_slot_cmp);
+
+    if (found == NULL) {
+        ngx_log_error(NGX_LOG_DEBUG, ngx_cycle->log, 0,
+            "Status code %ui not found in status_codes array", status);
+        return 0;
+    }
+
+    ngx_int_t index = (ngx_int_t)(found - (ngx_uint_t *)status_codes->elts);
+    return index + 1;
+}
```

Comment:
@wpjunior This function has no longer returned a negative value, we would like to return `ngx_uint_t` type, so that way, we can write redundant cast to keep code readability.
```suggestion
ngx_uint_t
ngx_http_vhost_traffic_status_find_status_code_slot(ngx_uint_t status, ngx_array_t *status_codes)
{
    ngx_uint_t *found = (ngx_uint_t *) bsearch(&status, status_codes->elts, status_codes->nelts,
        sizeof(ngx_uint_t),
        ngx_http_vhost_traffic_status_find_status_code_slot_cmp);

    if (found == NULL) {
        ngx_log_error(NGX_LOG_DEBUG, ngx_cycle->log, 0,
            "Status code %ui not found in status_codes array", status);
        return 0;
    }

    return found - (ngx_uint_t *)status_codes->elts + 1;
}
```